lstchain/scripts/lstchain_create_run_summary.py

```diff
@@ -0,0 +1,205 @@
+        dragon_reference_counter = combine_counters(
+            dragon_counters["pps_counter"][module_index],
+            dragon_counters["tenMHz_counter"][module_index],
+        )
+
+        ucts_available = bool(first_event.lstcam.extdevices_presence & 2)
+        run_start = int(round(Time(f.camera_config.date, format="unix").unix_tai)) * int(1e9)
+
+        if ucts_available:
+            if int(f.camera_config.lstcam.idaq_version) > 37201:
+                cdts = first_event.lstcam.cdts_data.view(CDTS_AFTER_37201_DTYPE)
+            else:
+                cdts = first_event.lstcam.cdts_data.view(CDTS_BEFORE_37201_DTYPE)
+
+            ucts_timestamp = cdts["timestamp"][0]
+            dragon_reference_time = ucts_timestamp
+            dragon_reference_source = "ucts"
+        else:
+            ucts_timestamp = -1
+            dragon_reference_time = run_start
+            dragon_reference_source = "run_start"
+
+        return dict(
+            ucts_timestamp=ucts_timestamp,
+            run_start=run_start,
+            dragon_reference_time=dragon_reference_time,
+            dragon_reference_module_id=module_id,
+            dragon_reference_module_index=module_index,
+            dragon_reference_counter=dragon_reference_counter,
+            dragon_reference_source=dragon_reference_source,
+        )
+    except AttributeError:
+        log.error(f"Files {pattern} do not contain events")
+
+        return dict(
+            ucts_timestamp=-1,
+            run_start=-1,
+            dragon_reference_time=-1,
+            dragon_reference_module_id=-1,
+            dragon_reference_module_index=-1,
+            dragon_reference_counter=-1,
+            dragon_reference_source=None,
+        )
+
+
+def main():
+    """
+    Write run summary to a file the following information per run:
+    Run number
+    Number of subruns
+    Type of run
+    Start_of_the_run
+    Event ID used to take time reference
+    Reference_source ("ucts" or "run_date")
+    Reference_timestamp
+    Initial dragon counter
+    Dragon module ID used to take the counter values
+    """
+    # TODO: Be able to create file incrementally run-by-run
+
+    args = parser.parse_args()
+
+    date_path = args.r0_path / args.date
+
+    list_of_files = get_list_of_files(date_path)
+    list_of_run_objects = get_list_of_runs(list_of_files)
+    run_numbers, n_subruns = get_runs_and_subruns(list_of_run_objects)
+    #    list_type_of_runs = [type_of_run(run) for run in run_numbers]
+    reference_counters = [read_counters(date_path, run) for run in run_numbers]
+
+    run_summary = Table(reference_counters)
+    run_summary.add_column(run_numbers, name="run_numbers", index=0)
+    run_summary.add_column(n_subruns, name="n_subruns", index=1)
+    #    run_summary.add_column(list_type_of_runs, name="type_of_run", index=2)
+    run_summary.write(args.output_dir / f"RunSummary_{args.date}.csv", format="ascii.csv")
```

Comment:
I'd recommend using ECSV format. It's basically the same, but with structured header information, so you can add metadata to the table, like the date, software version, etc.  (https://docs.astropy.org/en/stable/api/astropy.io.ascii.Ecsv.html).

Reply:
Thanks for the suggestion. I will change it.

Reply:
Done